test/Datadog.Trace.Tests/Configuration/ConfigurationSourceTests.cs

```diff
@@ -63,11 +63,16 @@ public static IEnumerable<object[]> GetTestData()
 
             yield return new object[] { ConfigurationKeys.Environment, "staging", CreateFunc(s => s.Environment), "staging" };
 
+            yield return new object[] { ConfigurationKeys.ServiceVersion, "1.0.0", CreateFunc(s => s.ServiceVersion), "1.0.0" };
+
             yield return new object[] { ConfigurationKeys.ServiceName, "web-service", CreateFunc(s => s.ServiceName), "web-service" };
+            yield return new object[] { "DD_SERVICE_NAME", "web-service", CreateFunc(s => s.ServiceName), "web-service" };
 
             yield return new object[] { ConfigurationKeys.DisabledIntegrations, "integration1;integration2", CreateFunc(s => s.DisabledIntegrationNames.Count), 2 };
 
             yield return new object[] { ConfigurationKeys.GlobalTags, "k1:v1, k2:v2", CreateFunc(s => s.GlobalTags.Count), 2 };
+            yield return new object[] { ConfigurationKeys.GlobalTags, "k1:,:,:v2,k3:v3", CreateFunc(s => s.GlobalTags.Count), 1 };
```

Comment:
Would it make sense to test that the tag here is actually `k3:v3` and not one of the other ones?

Reply:
Leaving this for a future PR.